sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java

```diff
@@ -3,36 +3,89 @@
 
 package com.azure.spring.cloud.autoconfigure.jms.properties;
 
+import com.azure.spring.core.implementation.connectionstring.ServiceBusConnectionString;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.boot.autoconfigure.jms.JmsPoolConnectionFactoryProperties;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
 import org.springframework.jms.support.QosSettings;
 import org.springframework.util.StringUtils;
 
-import javax.annotation.PostConstruct;
+import java.time.Duration;
 
 /**
  * {@link ConfigurationProperties} for configuring Azure Service Bus JMS.
  */
-@ConfigurationProperties(AzureServiceBusJmsProperties.PREFIX)
-// TODO(xiada): does this need to implement AzureProperties?
-public class AzureServiceBusJmsProperties {
+@ConfigurationProperties(prefix = AzureServiceBusJmsProperties.PREFIX)
+public class AzureServiceBusJmsProperties implements InitializingBean {
 
     public static final String PREFIX = "spring.jms.servicebus";
 
+    private static final String DEFAULT_REMOTE_URL = "amqp://localhost:5672";
+
+    private static final String AMQP_URI_FORMAT = "amqps://%s?amqp.idleTimeout=%d";
+
     private String connectionString;
 
     /**
      * JMS clientID. Only works for the bean of topicJmsListenerContainerFactory.
      */
     private String topicClientId;
 
-    private int idleTimeout = 1800000;
+    private Duration idleTimeout = Duration.ofMillis(1800000);
```

Comment:
How about changing it to be more readable, like Duration.ofMinutes?

Reply:
Yes, 30mins looks readable.